icu4c/source/i18n/uspoof_impl.cpp

```diff
@@ -196,6 +196,9 @@ void SpoofImpl::setAllowedLocales(const char *localesList, UErrorCode &status) {
     tmpSet = allowedChars.clone();
     const char *tmpLocalesList = uprv_strdup(localesList);
     if (tmpSet == NULL || tmpLocalesList == NULL) {
+        if (tmpLocalesList != NULL) {
```

Comment:
If tmpSet is NULL then there are no point to call  uprv_strdup(localesList); right?
How about we change them to  
```
tmpSet = allowedChars.clone();
if (tmpSet == nullptr) { 
status = U_MEMORY_ALLOCATION_ERROR; 
return; 
}
const char *tmpLocalesList = uprv_strdup(localesList);
if (tmpLocalesList == nullptr) { 
status = U_MEMORY_ALLOCATION_ERROR; 
return; 
}
```

Reply:
@FrankYFTang like this?:

```diff
--- a/icu4c/source/i18n/uspoof_impl.cpp
+++ b/icu4c/source/i18n/uspoof_impl.cpp
@@ -194,8 +194,12 @@ void SpoofImpl::setAllowedLocales(const char *localesList, UErrorCode &status) {
 
     // Store the updated spoof checker state.
     tmpSet = allowedChars.clone();
+    if (tmpSet == nullptr) {
+        status = U_MEMORY_ALLOCATION_ERROR;
+        return;
+    }
     const char *tmpLocalesList = uprv_strdup(localesList);
-    if (tmpSet == nullptr || tmpLocalesList == nullptr) {
+    if (tmpLocalesList == nullptr) {
         status = U_MEMORY_ALLOCATION_ERROR;
         return;
     }
```